frontend2/src/views/Login.tsx

```diff
@@ -1,7 +1,105 @@
-import React from "react";
+import React, { useContext, useEffect, useState } from "react";
+import Input from "../components/elements/Input";
+import Button from "../components/elements/Button";
+import { login as apiLogin } from "../utils/api/auth";
+import { type SubmitHandler, useForm } from "react-hook-form";
+import { FIELD_REQUIRED_ERROR_MSG } from "../utils/constants";
+import { Link, useNavigate } from "react-router-dom";
+import { EpisodeContext } from "../contexts/EpisodeContext";
+import { useCurrentUser, AuthStateEnum } from "../contexts/CurrentUserContext";
+import { type Maybe } from "../utils/utilTypes";
+import { getUserUserProfile } from "../utils/api/user";
+
+interface LoginFormInput {
+  username: string;
+  password: string;
+}
 
 const Login: React.FC = () => {
-  return <p>login page</p>;
+  const { register, handleSubmit } = useForm<LoginFormInput>();
+  const { episodeId } = useContext(EpisodeContext);
+  const { login, authState } = useCurrentUser();
+  const [loginError, setLoginError] = useState<Maybe<string>>();
+
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    // redirect to home if already logged in
+    if (authState === AuthStateEnum.AUTHENTICATED) {
+      navigate(episodeId !== undefined ? `/${episodeId}/home` : "/");
+    }
+  }, [authState]);
+
+  const onSubmit: SubmitHandler<LoginFormInput> = async (data) => {
+    try {
+      setLoginError(undefined);
+      await apiLogin(data.username, data.password);
+      login(await getUserUserProfile());
+    } catch (error) {
+      console.log(error);
+      setLoginError(
+        "Error logging in. Did you enter your username and password correctly?",
+      );
+    }
+  };
+  return (
+    <div className="flex h-screen flex-col items-center justify-center bg-gradient-to-tr from-cyan-200 to-cyan-700 p-2">
+      <div className="mb-6 flex flex-1 items-end text-center font-display text-5xl tracking-wide text-white sm:text-6xl">
+        BATTLECODE
+      </div>
+      {/* https://github.com/orgs/react-hook-form/discussions/8622 */}
+      <form
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        onSubmit={handleSubmit(onSubmit)}
+        className="flex w-11/12 flex-col gap-5 rounded-lg bg-gray-100 p-6 shadow-md sm:w-[350px]"
+      >
+        <h3 className="text-center text-xl font-light text-gray-700">
+          Log in to Battlecode
+        </h3>
+        {
+          // TODO: replace this with our custom notification component
```

Comment:
Maybe should make an issue for this so the TODO can track it?

Reply:
I'll put this into our roadmap spreadsheet